### PR TITLE
Removed container dependencies

### LINKED ISSUE #25: fix deprecation warnings (>=Symfony 4.2)
+ [ ] `The "Welp\MailchimpBundle\Command\SynchronizeMergeFieldsCommand" class extends "Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand" that is deprecated since Symfony 4.2, use {@see Command} instead.`
+ [ ] `The "Welp\MailchimpBundle\Command\SynchronizeSubscribersCommand" class extends "Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand" that is deprecated since Symfony 4.2, use {@see Command} instead.`
+ [ ] `The "Welp\MailchimpBundle\Command\WebhookCommand" class extends "Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand" that is deprecated since Symfony 4.2, use {@see Command} instead.`

### PR DESCRIPTION
**Pull request by jeroendk**
Closes #25
Related to #25 

Removed all the container dependencies and injected the needed services instead.
Because the list provider and the subscriber providers are dynamic, these services are set in the compilerPass. 

Let me know if you think we should do this some other way! 

I try to look into the Flex Recipe soon as well.

### CODE REVIEW

File: src/Command/SynchronizeSubscribersCommand.php [hunk 1]
```diff
@@ -2,17 +2,48 @@
...
+use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
-use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
 use Welp\MailchimpBundle\Provider\ListProviderInterface;
+use Welp\MailchimpBundle\Subscriber\ListSynchronizer;
 
-class SynchronizeSubscribersCommand extends ContainerAwareCommand
+class SynchronizeSubscribersCommand extends Command
```
Comment (Nightbr) [inline src/Command/SynchronizeSubscribersCommand.php:13]: Is this still compatible with SF2?

**Review by Nightbr — COMMENTED** [review]
Seems good to me but we need to be sure this will still be compatible with SF2 without BC before merging this.

**Comment by Nightbr** [comment]
Add to that, I think we should run phpunit tests in the CI .travis.yaml uncomment the line to run phpunit.

BTW thanks again for your contribution 🤟👌

File: src/Command/SynchronizeSubscribersCommand.php (hunk 1 above)
Comment (jeroendk) [reply src/Command/SynchronizeSubscribersCommand.php:13]: I will check.

**Comment by jeroendk** [comment]
Regarding the phpunit tests, the API key you provided is deactivated so they won't pass in Travis.

File: src/Command/SynchronizeSubscribersCommand.php (hunk 1 above)
Comment (jeroendk) [reply src/Command/SynchronizeSubscribersCommand.php:13]: Yes, the Command class exists in 2.8:
https://symfony.com/doc/2.8/console.html

The ContainerAwareCommand was for commands that needed the container.

**Comment by Nightbr** [comment]
yeah, we need to mock the underlying mailchimp-api in phpunit to be able to tests in travis. Perhaps we can do this in another PR.

For this one, seems good to me but I think we have not enough test to be 100% sure this will work in a Symfony project without any bug. I will try to tests this in a symfony project & if you can do the same it could be great and we will merge this.

Thanks in advance 👌

**Comment by jeroendk** [comment]
Great, I already tested all the commands in a sandbox project (Symfony 4.3). Did not find any new bugs.

**Comment by Nightbr** [comment]
Ok so here we go ;) thanks again for your contribution 🎉

**Comment by Nightbr** [comment]
Release on [1.3.1](https://github.com/welpdev/mailchimp-bundle/releases/tag/1.3.1)

**Pull request merged**